controller/lib/debug/vars.h

```diff
@@ -0,0 +1,121 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef VARS_H
+#define VARS_H
+
+#include "debug.h"
+
+// Defines the type of variable
+enum class VarType {
+  UNKNOWN = 0,
+  INT32 = 1,
+  UINT32 = 2,
+  FLOAT = 3,
+};
+
+// This class represents a variable that you can read/write using the
+// debug serial port.
+//
+// We give each such variable a name which the debugger command line will
+// use to access it.  We can also link it with a C++ variable who's value
+// it will read or write.
+//
+// The fmt string gives the debug interface a suggestion on the best way
+// to display the variable's data.
+class DebugVar {
+public:
+  // Variable of unusual type.  We expect the get/set functions to be overloaded
+  // @param help String that the Python code displays describing the variable.
+  // @param fmt printf style format string.  This is a hint to the Python code
+  // as to how the variable data should be displayed.
+  DebugVar(const char *name = "", const char *help = "", const char *fmt = "");
```

Comment:
Seems like the format should be something that I specify as the user in Python?  Like, maybe we have a floating-point value; maybe I want to see more or fewer decimal points.

Reply:
The idea is that we can give a good default format here.  Yes, it would be nicer if the user could specify a non-default format in get command.  I'll add that to the Python do_get function